### ISSUE TITLE
WorkUtil.teleportEntity conflict with BetterPortals

### ISSUE THREAD

**Issue by Johni0702** [micdoodle8/Galacticraft#3835]
1.12.2
https://github.com/Johni0702/BetterPortals/issues/421
https://paste.dimdev.org/hoduvudite

This is somewhat related to #2832 but this time with BetterPortals, which also completely revamps chunk management.
In particular, BP has a [hook into `PlayerList.transferPlayerToDimension`](https://github.com/Johni0702/BetterPortals/blob/9602911e69fa121066f662bd662fef17cd5030bc/src/view/java/de/johni0702/minecraft/view/impl/mixin/MixinPlayerList_NoSponge.java) right before the Respawn packet is sent, to allow it to tear down the world views it manages before the client switches the main world.

Looking at [the EntityPlayerMP code path](https://github.com/micdoodle8/Galacticraft/blob/a3b52c2f0bce5f711255f6e06b18b610874d6fe4/src/main/java/micdoodle8/mods/galacticraft/core/util/WorldUtil.java#L811-L866) of `WorldUtil.teleportEntity` (same might also apply to `teleportEntitySimple`, haven't yet looked at it but I noticed it also sends its own Respawn packet), I fail to see why it exists in the first place instead of just calling out to `PlayerList.transferPlayerToDimension` (except maybe for historical reasons). Is there anything it does which requires duplicating the code from transferPlayerToDimension?

This isn't even just a BP-specific issue, by duplicating the `transferPlayerToDimension` code, one can easily introduce subtle bugs when they get out of sync: Either because a mod like BP modifies it at runtime, or just due to oversight during updates (e.g. notice that `teleportEntity` never sends any `SPacketEntityProperties`, whereas `transferPlayerToDimension` does).

**Comment by radfast** [comment]
Hey @Johni0702  thanks for looking in.

>  I fail to see why it exists in the first place instead of just calling out to PlayerList.transferPlayerToDimension (except maybe for historical reasons)

I will need to take a detailed look to remember why it is done like this, might take me a few days.  It is _possible_ that it is for historical reasons - e.g. there wasn't a convenient hook in some earlier MC version so we had to create our own version.  It is also possible there's a good reason for it ...

**Comment by DarianLStephens** [comment]
Any word on this? I remember this was an issue before, and figured enough time went by to use the mods together, but it doesn't seem that's the case yet.

**Comment by Pi0nStEaM** [comment]
> Any word on this? I remember this was an issue before, and figured enough time went by to use the mods together, but it doesn't seem that's the case yet.

Yes, I too would like to know if this is safe to use with Galacticraft.

**Comment by DarianLStephens** [comment]
It currently isn't, but I've found Advanced Rocketry to be a refreshing take on space exploration, and appears to be entirely compatible with Better Portals, so I'll be using that for the foreseeable future.

**Comment by TheMCSpacePro** [comment]
So, this was just up and abandoned? Bumper.
